lib/full-site-editing/template-parts.php

```diff
@@ -13,6 +13,12 @@ function gutenberg_register_template_part_post_type() {
 		return;
 	}
 
+	// If the post type has already been registered (by WordPress core), skip
+	// registration.
+	if ( post_type_exists( 'wp_template_part' ) ) {
```

Comment:
I split this and the similar change in templates.php out into a separate PR (https://github.com/WordPress/gutenberg/pull/36898), but it's still needed here to be able to test alongside the WP core changes.

Reply:
This change breaks the site editor for WP 5.8. I also see unexpected behavior that James mentioned (https://github.com/WordPress/gutenberg/pull/36763#issuecomment-979177935) if I access the list page directly.

I think the reason is the difference between REST controllers, and since we have to support 5.8, it might be better to remove this check. What do you think?

## Screenshot
![CleanShot 2021-11-26 at 11 37 41](https://user-images.githubusercontent.com/240569/143544143-732b2246-376d-4f3a-b2f3-827be8cbadf4.png)

Reply:
This change breaks the site editor for WP 5.8. I also see unexpected behavior that James mentioned (https://github.com/WordPress/gutenberg/pull/36763#issuecomment-979177935) if I access the list page directly.

I think the reason is the difference between REST controllers, and since we have to support 5.8, it might be better to remove this check. What do you think?

## Screenshot
![CleanShot 2021-11-26 at 11 37 41](https://user-images.githubusercontent.com/240569/143544143-732b2246-376d-4f3a-b2f3-827be8cbadf4.png)

Reply:
> I also see unexpected behavior that James mentioned (#36763 (comment)) if I access the list page directly.

Do you see that only in 5.8?

I think we should consider updating the Gutenberg version of the REST API. Maybe we can also add a version check as part of this `if` statement.

Reply:
If I remove post type exists checks everything works correctly. The endpoint in WP 5.8 only supports user-created templates.

Reply:
Ok, I've reverted this for now, will move the conversation over to https://github.com/WordPress/gutenberg/pull/36898.

It does make it difficult to test this with the associated REST changes, best thing is to revert it back again if you want to do that.

Reply:
> I also see unexpected behavior that James mentioned (#36763 (comment)) if I access the list page directly.

I also haven't been able to repro this at all, not even in 5.8. Any additional details would be great.

Reply:
Were you able to reproduce Site Editor failure on 5.8? I will try to provide additional details.

Reply:
Yeah, I did reproduce that. Hopefully it's not a problem now that I've reverted the commit. Thanks for testing it 👍